sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.keyvault.spring;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.security.keyvault.secrets.SecretClient;
+import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+import com.azure.security.keyvault.secrets.models.SecretProperties;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Encapsulate key vault secret client in this class to provide a delegate of key vault operations.
+ */
+public class KeyVaultOperation {
+    private final long cacheRefreshIntervalInMs;
+    private final List<String> secretKeys;
+
+    private final Object refreshLock = new Object();
+    private final SecretClient keyVaultClient;
```

Comment:
Is there a reason to not use async clients here? Also, name the variable as `secretClient` as KeyVault has 4 different types of clients.

Reply:
According to the nature of the Spring property source, it will block until all responses are returned, so we use the sync client.

Reply:
Changed to secretClient.